### PR TITLE
SCMOD-3294 / SCMOD-3295: Update Dropwizard version

### PR DESCRIPTION
**Pull request by morvinshah4**
https://jira.autonomy.com/browse/SCMOD-3294
https://jira.autonomy.com/browse/SCMOD-3295

Developer Build:
http://sou-jenkins2.hpeswlab.net:8080/job/JobService/job/JobService~job-service~SCMOD-3294-js~CI/

### CODE REVIEW

**Comment by dermot-hardy** [comment]
Why are there two branches?  What is the difference between the [SCMOD-3294](https://github.com/JobService/job-service/tree/SCMOD-3294) branch and the [SCMOD-3294-js](https://github.com/JobService/job-service/tree/SCMOD-3294-js) branch?

**Comment by dermot-hardy** [comment]
Looking at it the [SCMOD-3294](https://github.com/JobService/job-service/tree/SCMOD-3294) one seems to be behind the [SCMOD-3294-js](https://github.com/JobService/job-service/tree/SCMOD-3294-js) one so I've just went on ahead and deleted it.

File: job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java [hunk 1]
```diff
@@ -486,9 +486,7 @@ public void testJobWithPrerequisiteJobsAndDelays() throws Exception
             // Database as Completed.
             context.getTestResult();
         }
-
-        Thread.sleep(5000); // Add short delay to allow J1 + J2 to complete.
-
+        Thread.sleep(10000); // Add short delay to allow J1 + J2 to complete.
```
Comment (dermot-hardy) [inline job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java]: The sleep time has doubled?

File: job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java [hunk 2]
```diff
@@ -771,8 +769,8 @@ private void testJobDeletion(final boolean useTaskDataObject) throws Exception {
     @Test
     public void testJobDeletionWithPrerequisiteJobs() throws Exception
     {
-        numTestItemsToGenerate = 2;                 // CAF-3677: Remove this on fix
-        testItemAssetIds = generateWorkerBatch();   // CAF-3677: Remove this on fix
+        //numTestItemsToGenerate = 2;                 // CAF-3677: Remove this on fix
+        //testItemAssetIds = generateWorkerBatch();   // CAF-3677: Remove this on fix
```
Comment (dermot-hardy) [inline job-service-acceptance-tests/src/integration-test/java/com/hpe/caf/jobservice/acceptance/JobServiceEndToEndIT.java]: Why have these lines been commented out?

File: job-service/pom.xml [hunk 3]
```diff
@@ -151,6 +151,7 @@
       <groupId>org.powermock</groupId>
       <artifactId>powermock-api-mockito</artifactId>
       <scope>test</scope>
+    <!--  <version>1.6.6</version> -->
```
Comment (dermot-hardy) [inline job-service/pom.xml]: Remove pointless comment.

File: job-service/pom.xml [hunk 4]
```diff
@@ -243,7 +242,6 @@
       <version>1.2</version>
     </dependency>
     <!-- End Logging -->
-
```
Comment (dermot-hardy) [inline job-service/pom.xml]: Remove pointless whitespace change.

File: pom.xml [hunk 5]
```diff
@@ -52,13 +47,55 @@
...
+                <version>${caf.worker-framework.version}</version>
+            </dependency>
             <dependency>
                 <groupId>com.github.workerframework</groupId>
                 <artifactId>worker-framework</artifactId>
                 <version>${caf.worker-framework.version}</version>
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+
```
Comment (dermot-hardy) [inline pom.xml]: Remove pointless whitespace change.

File: worker-jobtracking-container/pom.xml [hunk 6]
```diff
@@ -472,7 +472,7 @@
             <properties>
                 <worker.adminport>8081</worker.adminport>
                 <rabbitmq.node.port>5672</rabbitmq.node.port>
-                <rabbitmq.ctrl.port>15672</rabbitmq.ctrl.port>
+                <rabbitmq.ctrl.port>5673</rabbitmq.ctrl.port>
```
Comment (dermot-hardy) [inline worker-jobtracking-container/pom.xml]: Remove this change as it is clearly unrelated to this Jira.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

**Comment by morvinshah4** [comment]
Created the branch SCMOD-3294-js because of Jenkins build HTTP_PROXY issue on the older Jenkins page.

**Comment by morvinshah4** [comment]
Reduced the delay to 6 seconds in the previous commit

File: pom.xml [hunk 7]
```diff
@@ -52,6 +47,47 @@
...
+   Workaround apparent issue in dependencyManagement section of org.jdbi:jdbi3-parent:3.1.0
+   whereby Spring dependencies are declared optional
+-->
+            <dependency>
+                <groupId>org.springframework</groupId>
+                <artifactId>spring-framework-bom</artifactId>
+                <version>4.2.4.RELEASE</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
```
Comment (dermot-hardy) [inline pom.xml]: This workaround has been included in the WorkerFramework itself:
https://github.com/WorkerFramework/worker-framework/pull/68

File: pom.xml [hunk 8]
```diff
@@ -52,6 +47,47 @@
...
+            <dependency>
+                <groupId>org.powermock</groupId>
+                <artifactId>powermock-api-mockito</artifactId>
+                <version>1.6.4</version>
+            </dependency>
+            <dependency>
+                <groupId>com.github.workerframework</groupId>
+                <artifactId>worker-tracking-report</artifactId>
+                <version>${caf.worker-framework.version}</version>
+            </dependency>
```
Comment (dermot-hardy) [inline pom.xml]: Why have you had to add `worker-tracking-report`?  I don't see it being referenced in Job Service at all.

File: pom.xml [hunk 9]
```diff
@@ -52,6 +47,47 @@
...
+            <dependency>
+                <groupId>org.mockito</groupId>
+                <artifactId>mockito-all</artifactId>
+                <version>1.9.5</version>
+            </dependency>
+            <dependency>
+                <groupId>org.mockito</groupId>
+                <artifactId>mockito-core</artifactId>
+                <version>1.10.19</version>
+            </dependency>
```
Comment (dermot-hardy) [inline pom.xml:59]: Why are we using different versions of `mockito-all` and `mockito-core`?

**Review by dermot-hardy — CHANGES_REQUESTED** [review]

**Comment by morvinshah4** [comment]
Removed the spring-framework-bom as it is added in the worker-framework and other minor changes

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**